lgc/patch/RegisterMetadataBuilder.cpp

```diff
@@ -907,14 +920,21 @@ void RegisterMetadataBuilder::buildPsRegisters() {
 // Builds register configuration for compute/task shader.
 void RegisterMetadataBuilder::buildCsRegisters(ShaderStage shaderStage) {
   assert(shaderStage == ShaderStageCompute || shaderStage == ShaderStageTask);
-  auto entryPoint = m_pipelineShaders->getEntryPoint(shaderStage);
   if (shaderStage == ShaderStageCompute) {
+    Function *attribFunc = nullptr;
+    for (Function &func : *m_module) {
+      // Only entrypoint and amd_gfx functions may have the function attribute for workgroup id.
+      if (isShaderEntryPoint(&func) || (func.getCallingConv() == CallingConv::AMDGPU_Gfx)) {
+        attribFunc = &func;
+        break;
+      }
+    }
```

Comment:
Can you give a bit of background here? Is this related to raytracing indirect call support, where we may have a module that doesn't have an entry point?

Reply:
Actually, the test CsBPermuteWave64.lgc exposes a crash here which has a calling conversion of amdgpu_gfx. So I imitated the process in ShaderInputs::fixupUses to fix the issue.

Reply:
Okay. My main worry was what happens when we have multiple functions in a module. But this is about whether we have workgroup IDs, and with multiple functions we need to be clear about the ABI anyway during PatchEntryPointMutate. So I think this code is okay.

Reply:
Thanks.